### PR TITLE
Fixed split calls. see #16

### LINKED ISSUE #16: panic: runtime error: index out of range [1] with length 1
Hello,

I had a panic with the master branch (go 1.13.8):

```
go run cmd/m3u-filter/main.go -config=config.yaml

panic: runtime error: index out of range [1] with length 1

goroutine 1 [running]:
github.com/hoshsadiq/m3ufilter/logger.Get.func1(0xc0000984b0, 0x0, 0x83f49f, 0x3, 0xb7a0c0)
	/home/user/git/m3ufilter/logger/logger.go:21 +0x314
github.com/sirupsen/logrus.(*TextFormatter).Format(0xc000098320, 0xc0000e8070, 0x30, 0x7f9220, 0xc0000e6401, 0xc0000815c0, 0xc00009f9f0)
	/home/user/go/pkg/mod/github.com/sirupsen/logrus@v1.4.2/text_formatter.go:133 +0x53b
github.com/sirupsen/logrus.(*Entry).write(0xc0000e8070)
	/home/user/go/pkg/mod/github.com/sirupsen/logrus@v1.4.2/entry.go:255 +0xa1
github.com/sirupsen/logrus.Entry.log(0xc000097180, 0xc000081590, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/home/user/go/pkg/mod/github.com/sirupsen/logrus@v1.4.2/entry.go:231 +0x19d
github.com/sirupsen/logrus.(*Entry).Log(0xc0000e8000, 0x4, 0xc00009fc10, 0x1, 0x1)
	/home/user/go/pkg/mod/github.com/sirupsen/logrus@v1.4.2/entry.go:268 +0xeb
github.com/sirupsen/logrus.(*Entry).Logf(0xc0000e8000, 0xc000000004, 0x848ea5, 0x1d, 0xc00009fdb8, 0x2, 0x2)
	/home/user/go/pkg/mod/github.com/sirupsen/logrus@v1.4.2/entry.go:314 +0xe2
github.com/sirupsen/logrus.(*Logger).Logf(0xc000097180, 0x4, 0x848ea5, 0x1d, 0xc00009fdb8, 0x2, 0x2)
	/home/user/go/pkg/mod/github.com/sirupsen/logrus@v1.4.2/logger.go:145 +0x94
github.com/sirupsen/logrus.(*Logger).Infof(...)
	/home/user/go/pkg/mod/github.com/sirupsen/logrus@v1.4.2/logger.go:159
github.com/hoshsadiq/m3ufilter/m3u.GetPlaylist(0xc000081350, 0xb790c8, 0x0, 0x0, 0x4e8200)
	/home/user/git/m3ufilter/m3u/process.go:35 +0x22e
main.run(0x7ffded7c860d, 0xb, 0xc000010018, 0xc000010020)
	/home/user/git/m3ufilter/cmd/m3u-filter/main.go:42 +0x11e
main.main()
	/home/user/git/m3ufilter/cmd/m3u-filter/main.go:31 +0x26b
exit status 2
```

Println for some variables:
```
go run cmd/m3u-filter/main.go -config=config.yaml

f.File: /home/user/git/m3ufilter/m3u/process.go
basePkg: github.com/hoshsadiq/m3ufilter
strings.Split: [/home/user/git/m3ufilter/m3u/process.go]
```

### PR DESCRIPTION
**Pull request by Surfoo**

### CODE REVIEW

File: logger/logger.go [hunk 1]
```diff
@@ -17,9 +18,9 @@ func Get() *logrus.Logger {
 		logger.SetFormatter(&logrus.TextFormatter{
 			FullTimestamp: true,
 			CallerPrettyfier: func(f *runtime.Frame) (string, string) {
-				repopath := strings.Split(f.File, basePkg+"/")[1]
-				fnc := strings.Split(f.Function, basePkg+"/")[1]
-				return fmt.Sprintf("%s()", fnc), fmt.Sprintf("%s:%d", repopath, f.Line)
+				paths := strings.Split(f.File, "/")
+				return fmt.Sprintf("%s()", f.Function[len(basePkg)+1:]),
+					fmt.Sprintf("%s:%d", paths[5]+"/"+paths[6], f.Line)
```
Comment (hoshsadiq) [inline logger/logger.go]: This is wrong. In my case, my path is `/home/hosh/Workspace/src/github.com/hoshsadiq/m3ufilter`, thus paths becomes:
```
paths = {[]string} len:10, cap:10
 0 = {string} ""
 1 = {string} "home"
 2 = {string} "hosh"
 3 = {string} "Workspace"
 4 = {string} "src"
 5 = {string} "github.com"
 6 = {string} "hoshsadiq"
 7 = {string} "m3ufilter"
 8 = {string} "server"
 9 = {string} "server.go"
```
meaning, `paths[5]+"/"+paths[6]` becomes `github.com/hoshsadiq`, which is incorrect. In addition you can't base it on the last two items.

**Review by hoshsadiq — APPROVED** [review]

**Comment by hoshsadiq** [comment]
I've pushed a fix, will merge when the build is done.

**Comment by hoshsadiq** [comment]
@Surfoo actually before I merge, can you check that it works?

**Comment by Surfoo** [comment]
hum, I still have the issue: 

```
$ go run cmd/m3u-filter/main.go -config=config.yaml

panic: runtime error: index out of range [1] with length 1

goroutine 1 [running]:
github.com/hoshsadiq/m3ufilter/logger.Get.func1(0xc0000c04b0, 0x0, 0x82741a, 0x3, 0xb7ab40)
	/home/user/git/m3ufilter2/logger/logger.go:26 +0x314
```

And a Println of `f.File` and `appPath`:

```
f.File: /home/baptiste/go/pkg/mod/github.com/sirupsen/logrus@v1.4.2/entry.go
appPath: /home/baptiste/git/m3ufilter2
```

**Comment by hoshsadiq** [comment]
I'm unable to reproduce that. Can you ensure you pull your own latest master branch?

**Comment by Surfoo** [comment]
Yes, my master branch is up to date. 
I don't understand why `f.File` is about logrus...

**Comment by hoshsadiq** [comment]
Me either, are you able to clear your pkg dir and clone from scratch and see if that helps? If not, it would be good to see if you can debug it and see where that call comes from.

**Comment by hoshsadiq** [comment]
@Surfoo I've merged it for now as I'd like to keep these changes anyway, if it still hasn't been resolved please feel free to propose a solution or alternatively provide step by step instructions on how you're trying to compile it. Without reproducing these changes I'm unable to provide a fix.

**Pull request merged**